transport.go

```diff
@@ -0,0 +1,63 @@
+package goftp
+
+import (
+	"crypto/tls"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// Transport implements the http.RoundTripper interface.
+// Typical usage would be to register a Transport to handle
+// ftp:// and/or ftps:// URLs with http.Transport.RegisterProtocol.
+type Transport struct {
+	Config
+}
+
+// RoundTrip implements the http.RoundTripper interface to allow an http.Client
+// to handle ftp:// or ftps:// URLs.
+func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	config := t.Config
+	switch req.URL.Scheme {
+	default:
+		return nil, http.ErrSkipAltProtocol
+	case "ftp":
+	case "ftps":
+		if config.TLSConfig == nil {
+			config.TLSConfig = &tls.Config{}
+		}
+	}
+
+	if req.URL.User != nil {
```

Comment:
Are you intentionally overriding the config.User/Password if they are in the URL? If so, can you expand the docs to detail how any Config fields are potentially overridden by the Transport?

Reply:
Sure, will do.

Reply:
Done (62a7219 and 4c43d0e)